runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -203,6 +232,39 @@ private boolean checkCoreArchitecture() {
         return true;
     }
 
+    private boolean verifyPackageInfo(PackageInfo packageInfo,
+            String algorithm, String fingerprint) {
+        if (packageInfo.signatures == null) {
+            Log.d(TAG, "No signature in package info");
```

Comment:
Because the developer can do anything with the APK, this is just a common situation we have considered, but not a error what we can not handle here. What we need to do is letting the user know the integrity of the runtime library can not be verified, but not raising a fatal exception or something like that.